src/components/FormContainer.tsx

```diff
@@ -263,11 +296,19 @@ const FormListContainer = <A,>(
 
   return (
     <>
+      <Prompt
+        when={formState === FormState.Dirty}
```

Comment:
Did you run across `react-hook-form`'s `isDirty`? 

It should be in scope already, in the above line:

`const { reset, handleSubmit, formState: {isDirty} } = useFormContext()`

Maybe you already saw that and there's a reason it doesn't fit this situation?

Reply:
I went ahead and switched the implementation to use the existing `FormContext`'s `isDirty`, and it seems to work the same.